server/jetstream_tpm_test.go

```diff
@@ -0,0 +1,260 @@
+// Copyright 2024 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build windows
+
+package server
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/nats-io/nats.go"
+)
+
+// The tests in this file are not complete, but are a start to test the TPM
+// with default settings. For a complete test we would need to use a TPM
+// configured with an owner password and a SRK password. This is not
+// typically the default or setup in CI/CD environments.
+
+var jsTPMConfigPassword = `
+	listen: 127.0.0.1:-1
+	jetstream: {
+		store_dir: %q
+		tpm {
+			keys_file: %q
+			encryption_password: "s3cr3t!"
+		}
+	}`
+
+var jsTPMConfigBadPassword = `
+    listen: 127.0.0.1:-1
+	jetstream: {
+		store_dir: %q
+		tpm {
+			keys_file: %q
+			encryption_password: "garbage"
+		}
+	}`
+
+var jsTPMConfigPasswordPcr = `
+	listen: 127.0.0.1:-1
+	jetstream: {
+		store_dir: %q
+		tpm {
+			keys_file: %q
+			encryption_password: "s3cr3t!"
+			pcr: %d
+		}
+	}`
+
+var jsTPMConfigAllFields = `
+	listen: 127.0.0.1:-1
+	jetstream: {
+		store_dir: %q
+		tpm {
+			keys_file: %q
+			encryption_password: "s3cr3t!"
+			pcr: %d
+			srk_password: %q
+			cipher: "aes"
+		}
+	}`
+
+var jsTPMConfigInvalidBothOptionsSet = `
+	listen: 127.0.0.1:-1
+	jetstream: {
+		store_dir: %q
+		encryption_key: "foo"
+		tpm {
+			keys_file: "bar.json"
+			encryption_password: "s3cr3t!"
+		}
+	}`
+
+func getTPMFileName(t *testing.T) string {
+	return os.TempDir() + "/" + t.Name() + "/jskeys.json"
```

Comment:
Avoid `os.TempDir()` in tests and use `t.TempDir()` instead, that way they are cleaned up automatically when the test ends.

This will probably mean you also don't need the `clearAndGetStoreDir` and `os.Remove()` defers.

Reply:
Will fix...